src/vis_cpu/vis_cpu.py

```diff
@@ -317,8 +323,10 @@ def vis_cpu(
         v = A_s[:, :, beam_idx] * v[np.newaxis, np.newaxis, :]
 
         for i in range(len(antpos)):
+            # We want to take an outer product over feeds/antennas, contract over
+            # E-field components, and integrate over the sky.
             vis[:, :, t, i : i + 1, i:] = np.einsum(
-                "ijln,jkmn->iklm", v[:, :, i : i + 1].conj(), v[:, :, i:], optimize=True
+                "jiln,jkmn->iklm", v[:, :, i : i + 1].conj(), v[:, :, i:], optimize=True
```

Comment:
This does what you comment it should do, so I think it is right. However, would that mean we also should be specifying the shape of `vis` above as `vis = np.zeros((nfeed, nfeed, ntimes, nant, nant), dtype=complex_dtype)` instead of `vis = np.zeros((nax, nfeed, ntimes, nant, nant), dtype=complex_dtype)`?

Reply:
Good catch&mdash;that is what the shape of `vis` should be.